airflow/contrib/hooks/gcp_cloud_build_hook.py

```diff
@@ -0,0 +1,134 @@
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+
+    :param api_version: API version used (for example v1 or v1beta1).
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    _conn = None
+
+    def __init__(self, api_version="v1", gcp_conn_id="google_cloud_default", delegate_to=None):
+        super(CloudBuildHook, self).__init__(gcp_conn_id, delegate_to)
+        self.api_version = api_version
+
+    def get_conn(self):
+        """
+        Retrieves the connection to Cloud Functions.
+
+        :return: Google Cloud Function services object.
+        :rtype: dict
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build("cloudbuild", self.api_version, http=http_authorized, cache_discovery=False)
+        return self._conn
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def create_build(self, body, project_id=None):
+        """
+        Starts a build with the specified configuration.
+
+        :param body: The request body.
+            See: https://cloud.google.com/cloud-build/docs/api/reference/rest/Shared.Types/Build
+        :type body: dict
+        :param project_id: Optional, Google Cloud Project project_id where the function belongs.
+            If set to None or missing, the default project_id from the GCP connection is used.
+        :type project_id: str
+        :return: None
+        """
+        service = self.get_conn()
+
+        # Create build
+        response = (
+            service.projects()
+            .builds()
+            .create(projectId=project_id, body=body)
+            .execute(num_retries=NUM_RETRIES)
+        )
+
+        # Wait
+        operation_name = response["name"]
+        self._wait_for_operation_to_complete(operation_name=operation_name)
+
+        # Get result
+        build_id = response["metadata"]["build"]["id"]
+
+        result = (
+            service.projects()
+            .builds()
+            .get(projectId=project_id, id=build_id)
+            .execute(num_retries=NUM_RETRIES)
+        )
+
+        return result
+
+    def _wait_for_operation_to_complete(self, operation_name):
+        """
+        Waits for the named operation to complete - checks status of the
+        asynchronous call.
+
+        :param operation_name: The name of the operation.
+        :type operation_name: str
+        :return: The response returned by the operation.
+        :rtype: dict
+        :exception: AirflowException in case error is returned.
+        """
+        service = self.get_conn()
+        while True:
```

Comment:
Should we put a timeout here?

Reply:
There is a timeout for the duration of the entire operator. I think that there is no need to add another timeout. WDYT?

Reply:
https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/gcp_compute_hook.py#L290-L320
I found a similar code in project. I prefer to keep the consequences in the project.